cluster/cluster-authentication.yaml

```diff
@@ -0,0 +1,81 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: cluster-authentication
+spec:
+  params:
+    - name: name
+      description: name of the cluster
+      type: string
+    - name: type
+      description: type of the resource
+      type: string
+      default: cluster
+    - name: url
+      description: address of the cluster
+      type: string
+    - name: username
+      description: username for basic authentication to the cluster
+      type: string
+    - name: password
+      description: password for basic authentication to the cluster
+      type: string
+      default: ""
+    - name: cadata
+      description: contains PEM-encoded certificate authority certificates
+      type: string
+      default: ""
+    - name: clientKeyData
+      description: contains PEM-encoded data from a client key file for TLS
+      type: string
+      default: ""
+    - name: clientCertificateData
+      description: contains PEM-encoded data from a client cert file for TLS
+      type: string
+      default: ""
+    - name: namespace
+      description: default namespace to use on unspecified requests
+      type: string
+      default: ""
+    - name: token
+      description: bearer token for authentication to the cluster
+      type: string
+      default: ""
+    - name: insecure
+      description: to indicate server should be accessed without verifying the TLS certificate
+      type: string
+      default: "true"
+  workspaces:
+    - name: kubeconfigFile
+  steps:
+    - name: write
+      image: quay.io/diagrawa/kubeconfigwriter:latest
+      command: ["/kubeconfigwriter"]
+      args:
+        # passing the required json in the form of string to generate the kubeconfig file.
+        #
+        - -clusterConfig
+        - '{
+              "name":"$(params.name)",
+              "type":"$(params.type)",
+              "url":"$(params.url)",
+              "username":"$(params.username)",
+              "password":"$(params.password)",
+              "cadata":"$(params.cadata)",
+              "clientKeyData":"$(params.clientKeyData)",
+              "clientCertificateData":"$(params.clientCertificateData)",
+              "namespace":"$(params.namespace)",
+              "token":"$(params.token)",
+              "Insecure":$(params.insecure)
+            }'
+    - name: copy
```

Comment:
do we really need this copy step.
Could you make ur kubeconfigwriter logic write directly to `$(workspaces.kubeconfigFile.path)`

Reply:
Yes, It can be done by that way also.

Reply:
may be add an extra flag to `kubeconfigwriter` `--destination` 
if it is not set generate the default destination from name.

Reply:
Okay.